src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -521,6 +522,16 @@ public void WritePerfCounter(string counter)
             }
         }
 
+        string IKnobValueContext.GetVariableValueOrDefault(string variableName)
+        {
+            throw new Exception("RuntimeKnobSource not supported for Microsoft.VisualStudio.Services.Agent.HostContext");
```

Comment:
I am not sure I am a fan of embedding "RuntimeKnobSource" in this method. Should this method throw a generic "MethodNotSupported" exception and trap it in RuntimeKnobSource and rethrow there with more context?

Reply:
Agreed - good call. Updated to trap it more generically in CompositeKnobSource, that way any context can throw that exception for any knob type.

Reply:
I think the place for this would be in RuntimeKnobSource, around  context.GetVariableValueOrDefault, because this error is in fact unique to RuntimeKnobSource and that way you still get the good error message if you don't use it from a CompositeKnobSource.

Reply:
Makes sense - done